### PR TITLE
WIP: Filedescriptors

### PR DESCRIPTION
**Pull request by juliaogris**
looking for early input on how best to carve this up, 
wether to put filedescriptor.go in its own package and if test structure is ok.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,3 +3,10 @@ module github.com/alecthomas/protobuf
 go 1.17
 
 require github.com/alecthomas/participle/v2 v2.0.0-alpha7
+
+require (
+	github.com/google/go-cmp v0.5.5
+	google.golang.org/protobuf v1.27.1
+)
```
Comment (juliaogris) [inline go.mod:10]: :( Only used in tests, I tried without but couldn't make much sense of the FileDescriptor diffs.

File: parser/filedescriptor.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+			e, err := enum(e.Enum)
+			if err != nil {
+				return nil, err
+			}
+			fd.EnumType = append(fd.EnumType, e)
+		case e.Service != nil:
+		case e.Option != nil:
+		case e.Extend != nil:
+		default:
+			return nil, errors.New("cannot interpret Entry")
```
Comment (alecthomas) [inline parser/filedescriptor.go:49]: I would just panic here as this is an unrecoverable programming error.

File: parser/filedescriptor.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
+package parser
```
Comment (alecthomas) [inline parser/filedescriptor.go:1]: This should go in its own top-level package, such as `compiler`

**Comment by juliaogris** [comment]
will rework just parsing messages and scalar fields.

File: parser/filedescriptor.go (hunk 2 above)
Comment (alecthomas) [reply parser/filedescriptor.go:49]: In fact if all errors are these, I'd just remove error returns and panic everywhere.

File: parser/filedescriptor.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		er := &pb.EnumDescriptorProto_EnumReservedRange{
+			Start: &start,
+			End:   &start,
+		}
+		if r.End != nil {
+			end := int32(*r.End)
+			er.End = &end
+		}
+		if r.Max {
+			var end int32 = 2147483647 // tested with protoc 🤷‍♀️
```
Comment (alecthomas) [inline parser/filedescriptor.go:96]: `math.MaxInt32`

File: parser/parser_test.go [hunk 5]
```diff
@@ -63,3 +63,22 @@ func imports(from *Proto) []*Import {
 	}
 	return result
 }
+
+func TestProtofiles(t *testing.T) {
```
Comment (alecthomas) [inline parser/parser_test.go:67]: This is identical to `TestParser()` above?

**Pull request closed without merging**